examples/secure-tracing/README.md

```diff
@@ -0,0 +1,112 @@
+
+- Data Encryption with OTLP receiver supporting TLS. Transport Layer Security (TLS) is employed to encrypt traces in transit between the application and the OpenTelemetry (OTLP) endpoint, fortifying data security.
+- Client Authentication via [Envoy](https://www.envoyproxy.io/docs/envoy/latest/start/start), a high-performance proxy. 
+Even though we can configure OTLP receiver with mTLS for client authentication, authorization is not supported by OpenTelemetry Collector. This is one of the reasons that we use Envoy for client authentication. It allows us to easily add authorization, ensuring that only authorized clients can submit traces to the ingestion service.
+
+In this example, we also include a test via telementrygen: a tool provided from this repository for generating synthetic trace data, which helps verify the security features and overall functionality of the set up. 
+
+
+## Data Encryption via TLS
+The OpenTelemetry Collector has detailed [documentation](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configtls/README.md) on how to configure TLS. In this example, we enable TLS for receivers which leverages server configuration. 
+
+Example
+```
+receivers:
+  otlp:
+    protocols:
+      grpc:
+        endpoint: mysite.local:55690
+        tls:
+          cert_file: server.crt
+          key_file: server.key
+```
+
+## Client Authentication
+Envoy [sandbox for TLS](https://www.envoyproxy.io/docs/envoy/latest/start/sandboxes/tls) is a good source for reference. A few elements in following configuration are good to call out,
+- `require_client_certificate` is set true
+- The `matcher` under `validation_context` expects a client certificate with `group-id` as well as `tenat-id` which provides more granular control.
+
+```
+typed_config:
+    "@type": type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.DownstreamTlsContext
+    require_client_certificate: true
+    common_tls_context:
+    # h2 If the listener is going to support both HTTP/2 and HTTP/1.1.
+    alpn_protocols: "h2"
+    tls_certificates:
+    - certificate_chain: 
+        filename: "/etc/envoy.crt"
+        private_key: 
+        filename: "/etc/envoy.key"
+    validation_context:
+        match_typed_subject_alt_names:
+        - san_type: URI
+        matcher:
+            Match tenant by two level info: group id and tenant-id.
+            exact: "aprn:trace-client:certmgr:::group-x:/ig/5003178/uv/tenant-a"
+        trusted_ca:
+        filename: "/etc/ca.crt"
+```
+
+## Setup Environment
+### Generate Certificates
+To generate various self-signed certificates, including those for Envoy and the OpenTelemetry Collector receiver, as well as tracing client certificate, we utilize the widely renowned open-source tool [OpenSSL](https://www.openssl.org/source/). 
```

Comment:
Is any specific version of openssl required?

Reply:
No specific dependency on openssl version. I added a tested version.